sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1404,6 +1404,15 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     }
   }
 
+  override def visitIdentifier(ctx: IdentifierContext): String = withOrigin(ctx) {
+    val keyword = ctx.getText
+    if (ctx.ansiReserved() != null) {
+      throw new ParseException(
+        s"'$keyword' is reserved and you cannot use this keyword as an identifier.", ctx)
```

Comment:
hmmm, do we need to do it in this PR? I think this PR just changes the list of non-reserved/reserved keywords according to the config.

Reply:
ah..., ok. But, we have no behaivour change between ansi=true and ansi=false now?

Reply:
ok, I reverted the unrelated stuffs.